dataprofiler/profilers/profile_builder.py

```diff
@@ -1857,6 +1864,53 @@ def _merge_correlation_helper(corr_mat1, mean1, std1, n1,
 
         return corr_mat
 
+    def _update_chi2(self):
+        """
+        Calculates the p-value from a chi-squared test
+        for homogeneity between all categorical columns.
+        """
+        n_cols = len(self._profile)
+
+        # Fill matrix with nan initially
+        chi2_mat = np.full((n_cols, n_cols), np.nan)
+        # Compute chi_sq for each
+        for i in range(n_cols):
+            profile1 = self._get_categorical_profile(i)
```

Comment:
i would just define it within this function as i don't think _get_categorical_profile would be used anywhere else.